app/src/main/java/tech/pauly/findapet/shared/LocationHelper.kt

```diff
@@ -101,19 +102,8 @@ constructor(@ForApplication private val context: Context,
 
     @SuppressLint("MissingPermission")
     open fun fetchNewLocation(accuracy: Int, locationChangedListener: (location: Location) -> Unit) {
-        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
-        val criteria = Criteria().also { it.accuracy = accuracy }
-        locationManager.requestSingleUpdate(criteria, object : LocationListener {
-            override fun onLocationChanged(location: Location) {
-                locationChangedListener(location)
-            }
-
-            override fun onStatusChanged(provider: String, status: Int, extras: Bundle) {}
-
-            override fun onProviderEnabled(provider: String) {}
-
-            override fun onProviderDisabled(provider: String) {}
-        }, null)
+        val fusedLocationClient = LocationServices.getFusedLocationProviderClient(context)
+        fusedLocationClient.lastLocation.addOnSuccessListener(locationChangedListener)
```

Comment:
Added story to expand this to make a real fetch for location and not just use most recent https://www.pivotaltracker.com/story/show/158577086